packages/database/src/core/util/libs/parser.ts

```diff
@@ -174,17 +174,17 @@ export const parseDatabaseURL = function(
       colonInd = dataURL.length;
     }
 
-    const parts = host.split('.');
-    if (parts.length === 3) {
+    const hostWithoutPort = host.slice(0, colonInd);
+    if (hostWithoutPort.toLowerCase() === 'localhost') {
+      domain = 'localhost';
+    } else if (hostWithoutPort.split('.').length <= 2) {
+      domain = hostWithoutPort; // Domain is at least 2 parts.
+    } else {
+      const dotInd = host.indexOf('.');
+      domain = host.substring(dotInd + 1);
+      subdomain = host.substring(0, dotInd).toLowerCase();
```

Comment:
Nit: Flip two lines above to parse string consecutively.

Reply:
Done